src/main/java/com/puppycrawl/tools/checkstyle/checks/metrics/CyclomaticComplexityCheck.java

```diff
@@ -99,50 +99,166 @@
+ *       &amp;&amp; a == d) { // 4, &amp;&amp;
+ *       if (c == e) { // 5, if
+ *         fun2();
+ *       }
+ *     } else if (a == c // 6, if
+ *       || a == g) { // 7, ||
+ *       fun3();
+ *     } else if (g == h) { // 8, if
+ *       try {
+ *         fun4();
+ *       } catch (Exception e) { // 9, catch
+ *       }
+ *     } else {
+ *       switch(n) { // switch is not considered when switchBlockAsSingleDecisionPoint is false
+ *         case 1: // 10, case
+ *           fun1();
+ *           break;
+ *         case 2: // 11, case
+ *           fun2();
+ *           break;
+ *         case 3: // 12, case
+ *           fun3();
+ *           break;
+ *         default:
+ *           break;
+ *       }
+ *     }
+ *   }
+ * }
+ * </pre>
+ * <p>
+ * To configure the check with a threshold of 4 and check only for while and do-while loops:
+ * </p>
+ * <pre>
+ * &lt;module name="CyclomaticComplexity"&gt;
+ *   &lt;property name="max" value="4"/&gt;
+ *   &lt;property name="tokens" value="LITERAL_WHILE, LITERAL_DO"/&gt;
+ * &lt;/module&gt;
+ * </pre>
+ * <p>
+ * Example:
+ * </p>
+ * <pre>
+ * class CyclomaticComplexity {
+ *   // Cyclomatic Complexity = 5
+ *   public void foo() { // 1, function declaration
+ *     while (a &lt; b // 2, while
+ *       &amp;&amp; a &gt; c) {
+ *       fun();
+ *     }
+ *     if (a == b) {
+ *       if (a &lt; c) {
+ *       } else {
+ *         do { // 3, do
+ *           fun();
+ *         } while (d);
+ *       }
+ *     } else if (c == d) {
+ *       while (c &gt; 0) { // 4, while
+ *         fun();
+ *       }
+ *       do { // 5, do-while
+ *         fun();
+ *       } while (e);
+ *     }
+ *   }
+ * }
+ * </pre>
+ * <p>
+ * To configure the check to consider switch-case block as one decision point.
  * </p>
  * <pre>
  * &lt;module name="CyclomaticComplexity"&gt;
- *   &lt;property name="max" value="15"/&gt;
+ *   &lt;property name="switchBlockAsSingleDecisionPoint" value="true"/&gt;
  * &lt;/module&gt;
  * </pre>
  * <p>
+ * Example:
+ * </p>
+ * <pre>
+ * class CyclomaticComplexity {
+ *   // Cyclomatic Complexity = 11
+ *   public void foo() { // 1, function declaration
+ *     if (a == b) { // 2, if
+ *       fun1();
+ *     } else if (a == c // 3, if
+ *       &amp;&amp; a == d) { // 4, &amp;&amp;
+ *       if (c == e) { // 5, if
+ *         fun2();
+ *       }
+ *     } else if (a == c // 6, if
+ *       || a == g) { // 7, ||
+ *       fun3();
+ *     } else if (g == h) { // 8, if
+ *       try {
+ *         fun4();
+ *       } catch (Exception e) { // 9, catch
+ *       }
+ *     } else {
+ *       switch(n) { // 10, switch
+ *         case 1:
+ *           fun1();
+ *           break;
+ *         case 2:
+ *           fun2();
+ *           break;
+ *         default:
+ *           break;
+ *       }
+ *     }
+ *     j = a &gt; k ? b : c; // 11, ?
+ *   }
+ * }
+ * </pre>
+ * <p>
  * Explanation on how complexity is calculated (switchBlockAsSingleDecisionPoint is set to false):
```

Comment:
this example looks redundant

Reply:
so should i remove the redundant cases?

Reply:
I think yes, because other examples are pretty good at explaining how complexity is calculated

Reply:
But this example shows how violations change with change in property. So should i remove the whole example?

Reply:
should i remove both of the examples below this?